apps/web/lib/getTeamMemberEmailFromCrm.ts

```diff
@@ -0,0 +1,246 @@
+  const routingFormResponseIdAsNumber = Number(query[ROUTING_FORM_RESPONSE_ID_QUERY_STRING]);
+  const routingFormResponseId = isNaN(routingFormResponseIdAsNumber) ? null : routingFormResponseIdAsNumber;
+  return routingFormResponseId;
+}
+
+async function getAttributeRoutingConfig(
+  data:
+    | {
+        routingFormResponseId: number;
+        eventTypeId: number;
+      }
+    | {
+        route: Pick<LocalRoute, "attributeRoutingConfig">;
+      }
+) {
+  if ("route" in data) {
+    return data.route.attributeRoutingConfig ?? null;
+  }
+  const { routingFormResponseId, eventTypeId } = data;
+  const routingFormQuery = await prisma.app_RoutingForms_Form.findFirst({
+    where: {
+      responses: {
+        some: {
+          id: routingFormResponseId,
+        },
+      },
+    },
+    select: {
+      routes: true,
+    },
+  });
+  if (!routingFormQuery || !routingFormQuery?.routes) return null;
+  const parsedRoutes = routesSchema.safeParse(routingFormQuery.routes);
+
+  if (!parsedRoutes.success || !parsedRoutes.data) return null;
+
+  // Find the route with the attributeRoutingConfig
+  // FIXME: There could be multiple routes with same action.eventTypeId, we should actually ensure we have the chosenRouteId in here and use that route.
+  const route = parsedRoutes.data.find((route) => {
+    if ("action" in route) {
+      return route.action.eventTypeId === eventTypeId;
+    }
+  });
+
+  if (!route || !("attributeRoutingConfig" in route)) return null;
+  // Get attributeRoutingConfig for the form
+  const attributeRoutingConfig = route.attributeRoutingConfig;
+
+  if (!attributeRoutingConfig) return null;
+  return attributeRoutingConfig;
+}
+
+function getEnabledRoutingFormAppSlugFromQuery(query: ParsedUrlQuery) {
+  // Determine if a routing form enabled app is in the query. Then pass it to the proper handler
+  // Routing form apps will have the format cal.appSlug
+  let enabledRoutingFormApp;
+
+  for (const key of Object.keys(query)) {
+    const keySplit = key.split(".");
+
+    const appSlug = keySplit[1];
+
+    if (enabledAppSlugs.includes(appSlug)) {
+      enabledRoutingFormApp = appSlug;
+      break;
+    }
+  }
+
+  return enabledRoutingFormApp;
+}
+
+/**
+ * Uses the owner of the contact directly from CRM
+ */
+async function getOwnerEmailFromCrm(eventData: EventData, email: string): Promise<string | null> {
+  const crmContactOwnerEmail = await getCRMContactOwnerForRRLeadSkip(email, eventData.metadata);
+  if (!crmContactOwnerEmail) return null;
+  // Determine if the contactOwner is a part of the event type
+  const contactOwnerQuery = await prisma.user.findFirst({
+    where: {
+      email: crmContactOwnerEmail,
+      hosts: {
+        some: {
+          eventTypeId: eventData.id,
+        },
+      },
+    },
+  });
+  if (!contactOwnerQuery) return null;
+  return crmContactOwnerEmail;
+}
+
+/**
+ * Handles custom lookup field logic
+ */
+async function getTeamMemberEmailUsingRoutingFormHandler({
```

Comment:
`handleRoutingFormOption` renamed to this and some logic refactored out in their functions